channeld/channeld.c

```diff
@@ -2915,6 +2915,42 @@ static size_t calc_weight(enum tx_role role, const struct wally_psbt *psbt)
 	return weight;
 }
 
+/* Get the remote node amount after the splice */
+static struct amount_msat splice_remote_balance(struct peer *peer,
+					    enum tx_role our_role,
+					    const struct wally_psbt *psbt,
+					    int chan_output_index,
+					    int chan_input_index)
+{
+	s64 relative_remote_val;
+	/* Current value of the remote channel balance */
+	struct amount_msat push_value;
+
+	/* We calculcate the `push_value` to send to the
+	 * hsmd, that is the remote amount in the channel
+	 * after the splice. */
+	push_value = peer->channel->view[REMOTE].owed[REMOTE];
```

Comment:
I think there's a confusion between `push_val` (fundee value) and `remote_val` (non-local value).   `hsmd_ready_channel` wants `push_val`, but this function is actually calculating `remote_val`.  I think we need to change this function so that it actually calculates the fundee (acceptor) value and rename this to `splice_acceptor_value`.  (h/t @ksedgwic)

Reply:
Aaaaah! Nice catch I was chatting with Ken about it this morning (Italy morning) 

I was thinking that the `push_value` was the value that the other node (always the remote) was putting inside the channel.

So if the `push_value` is the fundee value we should use the opener = role == TX_INITIATOR trick. That ken mentioned in my DM this morning. Pushing it right now

Reply:
What do you think about this https://github.com/ElementsProject/lightning/pull/6746/commits/3fe70d0c5cbaaaf95694cbb96f1e71e8e4297fc2


P.S: I am really proud of how much documentation this code has, it simplifies a lot of the problem and how we are reading it. Thanks for all the good comments @devrandom @ksedgwic